internal/pkg/cli/env_delete.go

```diff
@@ -54,47 +53,72 @@ type DeleteEnvOpts struct {
-		}
-		opts.EnvName = envName
-	}
-	if opts.EnvProfile == "" {
-		profile, err := opts.prompt.Get(
-			fmt.Sprintf(fmtEnvDeleteProfilePrompt, color.HighlightUserInput(opts.EnvName)),
-			envDeleteProfileHelpPrompt,
-			nil, // no validation needed
-			prompt.WithDefaultInput("default"))
-		if err != nil {
-			return fmt.Errorf("prompt to get the profile name: %w", err)
+// Validate returns an error if the individual user inputs are invalid.
+func (o *DeleteEnvOpts) Validate() error {
+	if o.EnvName != "" {
+		if err := o.validateEnvName(); err != nil {
+			return err
 		}
-		opts.EnvProfile = profile
 	}
 	return nil
 }
 
-// Validate returns an error if the environment name does not exist in the project, or if there are live applications under the environment.
-func (opts *DeleteEnvOpts) Validate() error {
-	// Validate that the environment is stored in SSM.
-	env, err := opts.storeClient.GetEnvironment(opts.ProjectName(), opts.EnvName)
+// Ask prompts for fields that are required but not passed in.
+func (o *DeleteEnvOpts) Ask() error {
+	if err := o.askEnvName(); err != nil {
+		return err
+	}
+	if err := o.askProfile(); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Execute deletes the environment from the project by first deleting the stack and then removing the entry from the store.
+// If an operation fails, it moves on to the next one instead of halting the execution.
+// The environment is removed from the store only if other delete operations succeed.
+// Execute assumes that Validate is invoked first.
+func (o *DeleteEnvOpts) Execute() error {
+	if err := initDeleteEnvClientsFromProfile(o); err != nil {
+		return nil
+	}
+	if err := o.validateNoRunningApps(); err != nil {
+		return err
+	}
+
+	shouldDelete, err := o.shouldDelete(o.ProjectName(), o.EnvName)
 	if err != nil {
-		return fmt.Errorf("get environment %s metadata in project %s: %w", opts.EnvName, opts.ProjectName(), err)
+		return err
+	}
+	if !shouldDelete {
+		return nil
+	}
+
+	isStackDeleted := o.deleteStack()
+	if isStackDeleted { // TODO Add a --force flag that attempts to remove from SSM regardless.
+		// Only remove from SSM if the stack and roles were deleted. Otherwise, the command will error when re-run.
+		o.deleteFromStore()
 	}
-	opts.env = env
+	return nil
+}
 
-	// Look up applications by searching for cloudformation stacks in the environment's region.
-	// TODO We should move this to a package like "describe" similar to the existing "store" pkg.
-	stacks, err := opts.rgClient.GetResources(&resourcegroupstaggingapi.GetResourcesInput{
+// RecommendedActions is a no-op for this command.
+func (o *DeleteEnvOpts) RecommendedActions() []string {
+	return nil
+}
+
+func (o *DeleteEnvOpts) validateEnvName() error {
+	if _, err := o.storeClient.GetEnvironment(o.ProjectName(), o.EnvName); err != nil {
+		return fmt.Errorf("get environment %s metadata in project %s: %w", o.EnvName, o.ProjectName(), err)
```

Comment:
Should this error message be more direct since it's user facing when the user inputs an environment name that wasn't found or doesn't exist?